app/src/main/java/org/wordpress/pioupiou/postlist/PostListActivity.java

```diff
@@ -124,17 +200,99 @@ public void onDismissed() {
         sendButton.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
+                mNewPostContent = editText.getText().toString();
                 // Clear the saved text
                 editText.setText("");
                 tooltip.dismiss(true);
-                publishPost(editText.getText().toString());
+                publishPost();
             }
         });
     }
 
     // Publish post
-    private void publishPost(String text) {
-        Timber.i("Publishing post: " + text);
-        // TODO: Prepare the post and dispatch the action
+    private void publishPost() {
+        if (TextUtils.isEmpty(mNewPostContent)) {
+            ToastUtils.showToast(this, "Can't publish an empty post");
+            return;
+        }
+
+        // instantiate a new post, event handler will take care of setting the content and uploading it
+        PostStore.InstantiatePostPayload payload = new PostStore.InstantiatePostPayload(getSite(), false);
+        mDispatcher.dispatch(PostActionBuilder.newInstantiatePostAction(payload));
+    }
+
+    private boolean hasPostListFragment() {
+        return getPostListFragment() != null;
+    }
+
+    private PostListFragment getPostListFragment() {
+        Fragment fragment = getFragmentManager().findFragmentById(R.id.list);
+        if (fragment instanceof PostListFragment) {
+            return (PostListFragment) fragment;
+        } else {
+            return null;
+        }
+    }
+
+    private void showProgress(boolean show) {
+        View progress = findViewById(R.id.progress);
+        progress.setVisibility(show ? View.VISIBLE : View.GONE);
+    }
+
+    private void showError(String message) {
+        ToastUtils.showToast(this, message, ToastUtils.Duration.LONG);
+    }
+
+    /*
+     * called whenever the post list changes, such as when a fetch posts has completed
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPostChanged(PostStore.OnPostChanged event) {
+        if (event.causeOfChange == PostAction.FETCH_POSTS) {
+            mIsFetchingPosts = false;
+        }
+
+        if (!event.isError()) {
+            showPosts(getSite());
+        } else {
+            showError("OnPostChanged error - "
+                    + event.error.message
+                    + " (" + event.causeOfChange.toString() + ")");
+        }
+
+        mSwipeRefreshLayout.setRefreshing(false);
+    }
+
+    /*
+     * called when a new post has been instantiated - we use this to set the new post's content
+     * and actually publish it
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPostInstantiated(PostStore.OnPostInstantiated event) {
```

Comment:
This isn't enforced in any style (yet), but as a convention in FluxC (and WPAndroid integration) we import as specifically as possible - so this line (and others like it) would be `OnPostInstantiated` rather than `PostStore.OnPostInstantiated`.

The same would apply to e.g. `PostStore.InstantiatePostPayload` and any other FluxC inner classes.